### ISSUE TITLE
False positive 942190 normal german text

### ISSUE THREAD

**Issue by Phascolarctidae** [coreruleset/coreruleset#4289]
### Description

Rule 942190 (Detects MSSQL code execution and information gathering attempts) triggers a false positive when normal German text contains a quotation mark followed by an exclamation mark and a capital letter - for example: 

"Quote"! Next sentence

This results in a match on the substring "! N and causes an Inbound Anomaly Score of 5, leading to a 403 block at rule 949110 (Inbound Anomaly Score Exceeded). The pattern "! N is normal punctuation and capitalization in many languages (e.g. ending a quote with an exclamation mark, followed by a new sentence). Therefore, this match appears to be a language-related false positive.

### How to reproduce the misbehavior (-> curl call)

https://sandbox.coreruleset.org/?searchValue=%22Quote%22!%20Next%20Sentence

`curl -H "x-format-output: txt-matched-rules" "sandbox.coreruleset.org/?searchValue=%22Quote%22%21%20Next%20sentence"` 

### Your Environment

* CRS version (v4.19):
* Paranoia level setting (PL1) :

### Confirmation

[X ] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by franbuehler** [comment]
Thank you for reporting your issue, @Phascolarctidae!

I checked the rule and the regex and this part is matching:
https://github.com/coreruleset/coreruleset/blob/main/regex-assembly/942190.ra#L9 and https://github.com/coreruleset/coreruleset/blob/main/regex-assembly/942190.ra#L11.

Resulting in 
```
[\"'`]\s*!\s*[\"'`\w]
```

This is almost the same as in CRS3:  
```
[\"'`]\s*?!\s*?[\"'`\w]
``` 
(https://github.com/coreruleset/coreruleset/blob/v3.4/dev/util/regexp-assemble/data/942190.data#L26C1-L26C24). I just wanted to compare.

And the minimal strings that it matches are `"!N` and `"!"` for example.

But is `"Quote"! Next sentence` really natural German language? I think, "direkte Rede" would be: `"Quote!" Next sentence`. So the ! would be before the ". 
I don't want to be picky, but I wonder if this case really happens often.

**Comment by dkegel-fastly** [comment]
Both the crs3 and crs4 regexps you posted do match benign text occasionally in the real world.
Often?  No.
In a commercial WAF, this might be too often, but in the CRS world, it's probably more common for whoever's maintaining the web site to tune the problem away.
